plugin/authorizer_client_gcp.go

```diff
@@ -0,0 +1,81 @@
+package gcpauth
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/vault/helper/strutil"
+	compute "google.golang.org/api/compute/v1"
+	iam "google.golang.org/api/iam/v1"
+)
+
+var _ client = (*gcpClient)(nil)
+
+// gcpClient implements client and communicates with the GCP API. It is
+// abstracted as an interface for stubbing during testing. See stubbedClient for
+// more details.
+type gcpClient struct {
+	computeSvc *compute.Service
+	iamSvc     *iam.Service
+}
+
+func (c *gcpClient) InstanceGroups(ctx context.Context, project string, boundInstanceGroups []string) (map[string][]string, error) {
+	// map of zone names to a slice of instance group names in that zone.
+	igz := make(map[string][]string)
+
+	if err := c.computeSvc.InstanceGroups.
+		AggregatedList(project).
+		Fields("items/*/instanceGroups/name").
```

Comment:
We should probably use [filters](https://cloud.google.com/compute/docs/reference/rest/v1/instances/aggregatedList#filter) for name and zones  instead of having the pagination function?

Reply:
It's not possible due to a bug in the API: https://stackoverflow.com/questions/50452838/google-cloud-compute-instances-aggregatedlist-with-filter-fails

Reply:
We'd also still need the pagination function, even if we used filters. Filters are applied _after_ the response is generated, so you'd still need to paginate, potentially though a bunch of empty pages. For example, if there are 400 results, and you filter, the first 20 pages might be empty, since they _were_ filled prior to filtering.

But yea - you can test trying to filter in the docs online. It's broken. There's an internal ticket about it.

Reply:
wow what a BIG OLE BUNCH OF FUN :(
lgtm then